src/itsdangerous/serializer.py

```diff
@@ -10,13 +10,20 @@
 from .signer import _make_keys_list
 from .signer import Signer
 
+_TAnyStr = t.TypeVar("_TAnyStr", str, bytes, covariant=True)
 
-def is_text_serializer(serializer: t.Any) -> bool:
+
+class _PDataSerializer(t.Protocol[_TAnyStr]):
+    def loads(self, payload: str | bytes) -> t.Any: ...
+    def dumps(self, obj: t.Any, **kwargs: t.Any) -> _TAnyStr: ...
+
+
+def is_text_serializer(serializer: _PDataSerializer[t.Any]) -> bool:
```

Comment:
You could make this a `TypeGuard`, although it's probably better to wait for `TypeIs` support to arrive, since that will yield better results. Although you could make it `TypeGuard[_PDataSerializer[str]]` for now and add a comment to change it to `TypeIs` later.